app/models/shipit/task.rb

```diff
@@ -312,6 +312,16 @@ def permalink
       Shipit::Engine.routes.url_helpers.stack_task_url(stack, self)
     end
 
+    def commit_to_revert_to
+      previous_deployed_commit = stack.previous_successful_deploy_commit(id)
+
+      if !previous_deployed_commit || previous_deployed_commit == Shipit::Stack::NoDeployedCommit
```

Comment:
Any reason why we can't just do `previous_deployed_commit || since_commit` here?

Reply:
previous_deployed_commit may be an instance of NoDeployedCommit.

I am amiable to the idea of just not using NoDeployedCommit for any of this stuff, if you think that's a good idea; it would simplify things to just nil-logic which is nicer IMO.

Reply:
ah right.

Can we get rid of the `!previous_deployed_commit` case then? Or is this case expected sometimes too?

Reply:
Ah, I see that NoDeployedCommit overrides `blank?` so it may work for the logic you described already.

Reply:
If I remove NoDeployedCommit then this can become just
```ruby
def commit_to_revert_to
      stack.previous_successful_deploy_commit(id) || since_commit
end
```

Reply:
Latest deploy simplifies the conditional to just checking for `NoDeployedCommit`. I've rebased that into the first commit.